src/amo/components/FeaturedAddons/index.js

```diff
@@ -38,19 +59,39 @@ export class FeaturedAddonsBase extends React.Component {
     }
   }
 
+  loadDataIfNeeded(nextProps = {}) {
+    const { addonType: lastAddonType } = this.props;
+    const {
+      errorHandler, dispatch, loading, results,
+    } = { ...this.props, ...nextProps };
+
+    const nextParams = nextProps && nextProps.params ?
+      nextProps.params : {};
+    const nextAddonType = nextParams.visibleAddonType ?
+      apiAddonType(nextParams.visibleAddonType) : this.requestedAddonType();
+
+    if ((!results || nextAddonType !== lastAddonType) && !loading) {
+      dispatch(getFeatured({
+        addonType: nextAddonType,
+        errorHandlerId: errorHandler.id,
+      }));
+    }
+  }
+
   render() {
-    const { addonType, hasSearchParams, loading, results } = this.props;
+    const { errorHandler, hasSearchParams, loading, results } = this.props;
 
     return (
       <div className="FeaturedAddons">
+        {errorHandler.hasError() ? errorHandler.renderError() : null}
         <h2 className="FeaturedAddons-header">
-          {this.headerForAddonType(addonType)}
+          {this.headerForAddonType()}
```

Comment:
Well, the old code had a bug because the `addonType` function parameter wasn't used at all. I wanted to convert this to Flow (which would have exposed it) but I didn't have time.